numpy/distutils/system_info.py

```diff
@@ -1540,140 +1540,212 @@ def get_atlas_version(**config):
+                return {}
+            dict_append(info, libraries=[('fblas_src', info_src)])
+        return info
 
-        need_lapack = 0
-        need_blas = 0
-        info = {}
-        if atlas_info:
-            l = atlas_info.get('define_macros', [])
-            if ('ATLAS_WITH_LAPACK_ATLAS', None) in l \
-                   or ('ATLAS_WITHOUT_LAPACK', None) in l:
-                need_lapack = 1
-            info = atlas_info
+    def _get_info_lapack(self):
+        info = get_info('lapack')
+        if not info:
+            warnings.warn(LapackNotFoundError.__doc__, stacklevel=2)
+            info_src = get_info('lapack_src')
+            if not info_src:
+                warnings.warn(LapackSrcNotFoundError.__doc__, stacklevel=2)
+                return {}
+            dict_append(info, libraries=[('flapack_src', info_src)])
+        return info
 
-        else:
-            warnings.warn(AtlasNotFoundError.__doc__, stacklevel=2)
-            need_blas = 1
-            need_lapack = 1
-            dict_append(info, define_macros=[('NO_ATLAS_INFO', 1)])
-
-        if need_lapack:
-            lapack_info = get_info('lapack')
-            #lapack_info = {} ## uncomment for testing
-            if lapack_info:
-                dict_append(info, **lapack_info)
-            else:
-                warnings.warn(LapackNotFoundError.__doc__, stacklevel=2)
-                lapack_src_info = get_info('lapack_src')
-                if not lapack_src_info:
-                    warnings.warn(LapackSrcNotFoundError.__doc__, stacklevel=2)
-                    return
-                dict_append(info, libraries=[('flapack_src', lapack_src_info)])
-
-        if need_blas:
-            blas_info = get_info('blas')
-            if blas_info:
-                dict_append(info, **blas_info)
-            else:
-                warnings.warn(BlasNotFoundError.__doc__, stacklevel=2)
-                blas_src_info = get_info('blas_src')
-                if not blas_src_info:
-                    warnings.warn(BlasSrcNotFoundError.__doc__, stacklevel=2)
-                    return
-                dict_append(info, libraries=[('fblas_src', blas_src_info)])
+    def _calc_info_lapack(self):
+        info = self._get_info_lapack()
+        # TODO check whether this is actually necessary to warn about?
+        #      I think the warning is superfluous!
+        # warnings.warn(AtlasNotFoundError.__doc__, stacklevel=2)
 
-        self.set_info(**info)
-        return
+        # Ensure we have NO_ATLAS_INFO in the macro list
+        dict_append(info, define_macros=[('NO_ATLAS_INFO', 1)])
+        info_blas = self._get_info_blas()
+        dict_append(info, **info_blas)
+        if info:
+            self.set_info(**info)
+            return True
+        return False
+
+    def calc_info(self):
+        # Copy list so we can overwrite
+        lapack_order = self.lapack_order[:]
+
+        user_order = os.environ.get('NUMPY_LAPACK_ORDER', None)
```

Comment:
better `NPY_LAPACK_ORDER` for consistency with other envvars (`NPY_RELAXED_STRIDES_CHECKING`, `NPY_DISTUTILS_APPEND_FLAGS`, `NPY_PKG_CONFIG_PATH`)

Reply:
Correct! I'll fix!